### PR TITLE
feat(Pipeline Migration to 1ES Template): migrated signed build to 1ES micro build template

### PR DESCRIPTION
**Pull request by v-viyada**
#### Details

This PR is for migrating Microsoft.accessibility-insights-windows(signed) pipeline to 1ES MicroBuild Template.

Old run link: https://dev.azure.com/mseng/1ES/_build/results?buildId=25829770&view=results
New run link: https://dev.azure.com/mseng/1ES/_build/results?buildId=25906954&view=results

Below are details:
1. We have to keep running compliance jobs as we are running TSA upload task. Currently 1ES template do not support dynamic TSA config file.
2. Binskim task is running twice in Signed Release job. One is auto injected by 1ES template and another one we added manually for TSA Upload. 1ES template do not support TSA upload for specific job in the pipeline. If we enable TSA in the template than it will inject TSA task in all the jobs including source SDL job. We cannot enable TSA for sourceSDL job because it require to hardcore tsa config in the repo.
3. Compared the output artifacts between old and new run. Below are the differences. There is no difference in the folders/files used in release pipelines AIWin: Automated Canary Release and AIWin: Release Promotion
   - _manifest folder in all the artifacts. This contains SBOM Manifest for the artifact. For Official template (used for production pipeline), this task is enabled by default.
   - DebugDrop has extra files as there are more SDL task running due to auto injection from the template.
   - MicroBuild folder missing from drop artifacts. This is because MicroBuild plugins are now installed using auto injected task which is define to download the plugins in Temporary directory instead of source directory. I tried copying them into artifacts but credscan task for artifact folders fails as this contains some secret. Also we are not using this into our release pipeline so it is not a required output. Pipeline run [link](https://dev.azure.com/mseng/1ES/_build/results?buildId=25454449&view=results) for credscan task failure.
5. ##[warning]1ES PT Warning: Failed to fetch default repository branch.
This is an existing issue. Please refer
[Bug 2119901](https://dev.azure.com/mseng/1ES/_workitems/edit/2119901): Address issue "Failed to fetch default branch"

##### Motivation

[User Story 2136513](https://dev.azure.com/mseng/1ES/_workitems/edit/2136513): Migrate Microsoft.accessibility-insights-windows(signed) to 1ES template

#### Pull request checklist

- [na] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/main/docs/Scenarios.md) completed?
- [na] Does this address an existing issue? If yes, Issue# - 
- [na] Includes UI changes?
  - [na] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [na] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

**Comment by DaveTryon** [comment]
@v-viyada, the presence of the `_manifest` folder in the `msi` artifact will break the Canary release script because it only expects a single folder to exist under the `msi` folder. To fix this, edit https://github.com/microsoft/accessibility-insights-windows/blob/main/tools/scripts/pipeline/deploy/Canary/set_variables.ps1#L18 from

```
$a11yInsightsVersion = Get-ChildItem -Name -Directory $($msiFolder)
```

to

```
$a11yInsightsVersion = Get-ChildItem -Name -Directory $($msiFolder) -Exclude '_manifest'
```

File: build/signedbuild.yml [hunk 1]
```diff
@@ -11,335 +11,360 @@ variables:
...
+        - task: CopyFiles@2
+          displayName: 'Copy Files for Signing Validation'
+          inputs:
+            SourceFolder: '$(Build.SourcesDirectory)\src'
+            Contents: |
+             AccessibilityInsights\bin\Release\net48\**\?(*.exe|*.dll)
+             AccessibilityInsights.VersionSwitcher\bin\Release\net48\**\?(*.exe|*.dll)
+             !AccessibilityInsights.VersionSwitcher\bin\Release\net48\**\?(Microsoft.Deployment.WindowsInstaller.dll|Newtonsoft.Json.dll)
+             MSI\bin\Release\msi\**\AccessibilityInsights.msi
+             Manifest\bin\Release\net48\AccessibilityInsights.Manifest.dll
```
Comment (DaveTryon) [inline build/signedbuild.yml:369]: It looks like the indents of lines 365-368 are slightly off. Should we fix them  while we're touching this file or put it into a separate PR?

Comment (v-viyada) [reply build/signedbuild.yml]: corrected the indentation

**Comment by v-viyada** [comment]
> @v-viyada, the presence of the `_manifest` folder in the `msi` artifact will break the Canary release script because it only expects a single folder to exist under the `msi` folder. To fix this, edit https://github.com/microsoft/accessibility-insights-windows/blob/main/tools/scripts/pipeline/deploy/Canary/set_variables.ps1#L18 from
> 
> ```
> $a11yInsightsVersion = Get-ChildItem -Name -Directory $($msiFolder)
> ```
> 
> to
> 
> ```
> $a11yInsightsVersion = Get-ChildItem -Name -Directory $($msiFolder) -Exclude '_manifest'
> ```

@DaveTryon updated the script to exclude _manifest folder

**Review by DaveTryon — APPROVED** [review]
Thanks, @v-viyada. Looks great!
